middleware-jackson/src/main/java/com/nytimes/android/external/store3/middleware/jackson/JacksonParserFactory.kt

```diff
@@ -0,0 +1,129 @@
+package com.nytimes.android.external.store3.middleware.jackson
+
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.nytimes.android.external.store3.base.Parser
+
+import java.io.Reader
+import java.lang.reflect.Type
+
+import okio.BufferedSource
+
+/**
+ * Factory which returns various Jackson [Parser] implementations.
+ */
+object JacksonParserFactory {
+
+    /**
+     * Returns a new Parser which parses from a String to the specified type, using
+     * the provided [JsonFactory] instance.
+     */
+    fun <T> createStringParser(jsonFactory: JsonFactory, type: Type): Parser<String, T> {
+        if (jsonFactory == null) {
```

Comment:
We probably don't need any of these null checks anymore

Reply:
@ramonaharrison Yes, although it depends on what the framework will return in such case. If `jsonFactory` is something that can be null but it is mistakenly not annotated, it should cause a crash with a `KotlinNullPointerException`. So it is either needed to be marked as nullable and retain the check, or remove the check :)

Reply:
Same goes for more null-checking in that file as well!

Reply:
I would go for avoiding null-checks. I prefer a "fail fast" strategy :-)